src/DataStreams/RemoteQueryExecutor.cpp

```diff
@@ -385,6 +380,14 @@ bool RemoteQueryExecutor::setPartUUIDs(const std::vector<UUID> & uuids)
     return true;
 }
 
+void RemoteQueryExecutor::processReadTaskRequest()
+{
+    if (!task_iterator)
+        throw Exception("Distributed task iterator is not initialized", ErrorCodes::LOGICAL_ERROR);
```

Comment:
I think this is also not always `LOGICAL_ERROR`. Remote server might send `ReadTaskRequest` packet when we don't expect (in case of bug, or version incompatibility).
But maybe ok for now.

Reply:
We have something similar for DDL for example 
https://github.com/ClickHouse/ClickHouse/blob/ee8b3339a11adc2e76bdc199bd188934fb3fdafe/src/Interpreters/Context.cpp#L1477